libsoftwarecontainer/src/gateway/devicenode/devicenodegateway.cpp

```diff
@@ -32,14 +33,28 @@ DeviceNodeGateway::DeviceNodeGateway() :
 ReturnCode DeviceNodeGateway::readConfigElement(const json_t *element)
 {
     DeviceNodeParser parser;
+    DeviceNodeLogic logic;
     DeviceNodeParser::Device dev;
 
     if (isError(parser.parseDeviceNodeGatewayConfiguration(element, dev))) {
         log_error() << "Could not parse device node configuration";
         return ReturnCode::FAILURE;
     }
 
-    m_devList.push_back(dev);
+    auto item = std::find_if(m_devList.begin(), m_devList.end(),
+        [&] (DeviceNodeParser::Device const &d) { return d.name == dev.name; });
```

Comment:
Consider defining equality for `DeviceNodeParser::Device` as being equality on the name parameter instead.